src/main/java/org/testcontainers/containers/TarantoolCartridgeContainer.java

```diff
@@ -451,21 +456,54 @@ protected void containerIsStarting(InspectContainerResponse containerInfo) {
     }
 
     private boolean setupTopology() {
-        try {
-            executeScript(topologyConfigurationFile).get();
-            // The client connection will be closed after that command
-        } catch (Exception e) {
-            if (e instanceof ExecutionException) {
-                if (e.getCause() instanceof TimeoutException) {
-                    return true;
-                    // Do nothing, the cluster is reloading
-                } else if (e.getCause() instanceof TarantoolConnectionException) {
-                    // Probably cluster is not ready
-                    logger().error("Failed to setup topology: {}", e.getMessage());
-                    return false;
+        if (topologyConfigurationFile == null) {
+            String runDirPath = null;
+            try {
+                Container.ExecResult envVariablesContainer = this.execInContainer("env");
+                String stdout = envVariablesContainer.getStdout();
+                int exitCode = envVariablesContainer.getExitCode();
+                if (exitCode != 0) {
+                    logger().error("Failed to bootstrap replica sets topology: {}", stdout);
+                }
+                int startInd = stdout.lastIndexOf(ENV_TARANTOOL_RUNDIR + "=");
+                try {
+                    runDirPath = stdout.substring(startInd,
+                            stdout.indexOf('\n', startInd)).split("=")[1];
+                } catch (Exception e) {
+                    logger().error("Missing dir-run environment variable: {}", e.getMessage());
+                }
+
+            } catch (Exception e) {
+                logger().error("Failed to get environment variables: {}", e.getMessage());
+            }
+
+            try {
+                Container.ExecResult lsResult = this.execInContainer("cartridge", "replicasets", "--run-dir=" + runDirPath, "setup", "--bootstrap-vshard");
```

Comment:
Need to pass the file name here. It's configurable as well as instances.yml.
```
 cartridge replicasets setup --help
Set up replica sets described in a file

Usage:
  cartridge replicasets setup [flags]

Flags:
      --file string        File where replica sets configuration is described
                           Defaults to replicasets.yml
      --bootstrap-vshard   Bootstrap vshard
      --name string        Application name
      --run-dir string     Directory where PID and socket files are stored
                           defaults to ./tmp/run ("run-dir" in .cartridge.yml)
      --cfg string         Configuration file for instances
```

Reply:
Corrected